busyApp/src/devBusyAsyn.c

```diff
@@ -195,40 +200,58 @@ static void processCallback(asynUser *pasynUser)
     if(pr->pact) callbackRequestProcessCallback(&pPvt->callback,pr->prio,pr);
 }
 
+/* This function is called from the general purpose callback task to process
+ * the record in response to a driver callback.  In this case the record should be
+ * processed but the driver should not be called
+ */
+static void driverProcessCallback(CALLBACK *pcb)
+{
+    devBusyCBStr *pCBStr;
+    callbackGetUser(pCBStr, pcb);
+    {
+        dbCommon *pr = pCBStr->pr;
+        devBusyPvt *pPvt = (devBusyPvt *)pr->dpvt;
+        struct rset *prset = (struct rset *)pr->rset;
+        dbScanLock(pr);
+        pPvt->callbackValue = pCBStr->value;
+        pPvt->newCallbackValue = 1;
+        (prset->process)(pr);
+        dbScanUnlock(pr);
+        free(pCBStr);
+    }
+}
 
 static void interruptCallback(void *drvPvt, asynUser *pasynUser,
                 epicsInt32 value)
 {
     devBusyPvt *pPvt = (devBusyPvt *)drvPvt;
     busyRecord *pr = (busyRecord *)pPvt->pr;
+    devBusyCBStr *pCBStr;
 
     if (!interruptAccept) return;
-    dbScanLock((dbCommon *)pr);
-    asynPrint(pPvt->pasynUser, ASYN_TRACEIO_DEVICE,
-        "%s devBusyAsyn::interruptCallback pr->val=%d, new value=%d\n",
-        pr->name, pr->val, value);
-    pPvt->callbackValue = value;
-    pPvt->newCallbackValue++;
-    scanOnce((dbCommon *)pr);
-    dbScanUnlock((dbCommon *)pr);
+    pCBStr = (devBusyCBStr *) malloc(sizeof(devBusyCBStr));
```

Comment:
@MarkRivers is this really calling `malloc()` and `free()` for every single value sent through the device support? Apart from that being a major speed bump it could have an unfortunate effect on the lifetime of the IOC (it will definitely cause problems on VxWorks 5.5.x systems, hopefully less so with VxWorks 6.x as they replaced the memory allocator there to one which doesn't fragment the memory pool).

@timmmooney You may have been a bit quick on the Merge trigger here, I'm not sure this is quite ready.